akka-actor/src/main/scala/akka/pattern/BackoffOnRestartSupervisor.scala

```diff
@@ -39,20 +39,37 @@ private class BackoffOnRestartSupervisor(
         // directive match {
         case Restart ⇒
           val childRef = sender()
-          become({
-            case Terminated(`childRef`) ⇒
-              unbecome()
-              child = None
-              val restartDelay = BackoffSupervisor.calculateDelay(restartCount, minBackoff, maxBackoff, randomFactor)
-              context.system.scheduler.scheduleOnce(restartDelay, self, BackoffSupervisor.StartChild)
-              restartCount += 1
-            case _ ⇒ // ignore
-          }, discardOld = false)
+          become(waitChildTerminatedBeforeBackoff(childRef) orElse handleBackoff)
```

Comment:
yes, and also explicitly ignoring the StartChild during this period. The rest of it are external management messages.